addons/website/static/tests/tours/link_tools.js

```diff
@@ -71,6 +71,12 @@ wTourUtils.registerWebsitePreviewTour('link_tools', {
         content: "Click on the secondary style button.",
         trigger: '#toolbar we-button[data-value="secondary"]',
     },
+    {
+        // Make sure the style has had time to be applied before saving
+        content: "The link should have the secondary button style.",
+        trigger: 'iframe .s_text_image a.btn.btn-secondary[href="http://odoo.be"]:contains("odoo website")',
+        run: () => {}, // It's a check.
+    },
```

Comment:
Apparently, the race condition appeared already in 16.0
![image](https://github.com/odoo/odoo/assets/22252071/bf30711d-1600-42f0-ab82-6d1df9e48f7b)
I don't remember why I targeted 16.1. So I have to investigate on the 16.0 or/and 15.2 branch.

Reply:
You probably targeted 16.1 because GDI's recent fix on that tour targeted 16.0 and we confirmed with a multi build that there were no issues over there anymore. But since you are saying that this is a very rare issue, we could have been wrong.

Need to retest with a multi build many times on a 16.0 branch with that test enabled.

If you are stuck, we can maybe try to ask to someone with fresh eyes to have a look, you tell me, don't want you to go crazy for no reason :wink:

Reply:
Yes, I'm quite stuck here. I'll ask someone to have a look